pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -326,6 +326,10 @@ func (alloc *Action) predicate(task *api.TaskInfo, node *api.NodeInfo) error {
 		statusSets = append(statusSets, &api.Status{Code: api.Unschedulable, Reason: api.WrapInsufficientResourceReason(resources)})
 		return api.NewFitErrWithStatus(task, node, statusSets...)
 	}
+	if node.Allocatable.MaxTaskNum <= len(alloc.session.NodeMap[node.Name].Pods) {
```

Comment:
Bacause it may influence the predicate result if there are too many pod running on the node, so I put it before the new predicate cache to make the cache accurate.

Reply:
Sorry I didn't get the point, you didn't remove these codes in the predicate plugin, so why do we need to add duplicate codes here?